### ISSUE TITLE
Manually added headers are reset in response

### ISSUE THREAD

**Issue by imalsogreg** [haskell-servant/servant-snap#17]
`wrapCors` from `snap-cors` for example monadically adds headers to responses, and `servant-snap` drops them.

**Comment by hvr** [comment]
For the record, a concrete example is https://github.com/hvr/hackage-matrix-builder/blob/ba7a671cf260900093374ff6ed8669a56bc8d5c9/src-exe/Controller/WebSvc.hs#L99-L102

**Comment by imalsogreg** [comment]
Using commit 72a42122614735d38c5d4319e6c7098af05f547a and the following test code, CORS headers are no longer dropped. I'm still confused though because a new CORS testcase won't pass.

```
{-# LANGUAGE DataKinds, OverloadedStrings #-}

module Tmp where

import Snap.CORS
import Data.Proxy
import Servant.API
import Servant.Server
import qualified Snap.Snaplet as S

type API = Get '[JSON] Int

api :: Proxy API
api = Proxy :: Proxy API

r :: IO ()
r = S.serveSnaplet  mempty $ S.makeSnaplet "" "" Nothing $
    wrapCORS >> S.addRoutes  [("", serveSnap api (return 100))] >> return ()
```

```
curl -v -H "Origin: http://example.com" localhost:8000/
*   Trying ::1...
* TCP_NODELAY set
* Connection failed
* connect to ::1 port 8000 failed: Connection refused
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8000 (#0)
> GET / HTTP/1.1
> Host: localhost:8000
> User-Agent: curl/7.54.0
> Accept: */*
> Origin: http://example.com
>
< HTTP/1.1 200 OK
< server: Snap/1.0.3.1
< access-control-allow-credentials: true
< access-control-allow-origin: http://example.com
< content-type: application/json;charset=utf-8
< date: Wed, 20 Sep 2017 01:18:51 GMT
< transfer-encoding: chunked
<
* Connection #0 to host localhost left intact
100%
```

**Comment by hvr** [comment]
@imalsogreg does it make any difference that I use `applyCORS` rather than `wrapCORS`? You may want to add a "control group" by adding another route entry in `addRoutes` which manually sets some header and then writes out some dummy text body (i.e. not using `serveSnap`)

**Comment by imalsogreg** [comment]
Thanks for the ideas :) I've added `applyCORS` to one of the two branches in `addRoutes` in the `greet.hs` example (will do this in the test suite when I figure out why my CORS attempts in the test suite are failing).

In summary, this issue is closed by 72a42122614735d38c5d4319e6c7098af05f547a
